lib/_stream_readable.js

```diff
@@ -1070,13 +1074,64 @@ Readable.prototype.wrap = function(stream) {
 };
 
 Readable.prototype[SymbolAsyncIterator] = function() {
+  if (typeof this.read === 'function') {
+    const iter = createAsyncIterator(this);
+    iter.stream = this;
+    return iter;
+  }
+
   if (createReadableStreamAsyncIterator === undefined) {
     createReadableStreamAsyncIterator =
       require('internal/streams/async_iterator');
   }
+
   return createReadableStreamAsyncIterator(this);
 };
 
+async function* createAsyncIterator(stream) {
+  let _resolve = nop;
+  let _reject = nop;
+  let _error = null;
+  let _ended = false;
+  let _closed = !!stream._readableState?.closeEmitted;
+
+  stream
+    .on('readable', () => {
+      _resolve();
```

Comment:
I think I might be misunderstanding things - what happens here if `readable` and `error` are emitted in the same tick?

`_error` would be set, `resolve` would be called (on readable) before reject, so the `_reject(_error)` call wouldn't get anywhere, we get to the `while` again, error is set, so we exit and call the destroyer with the error (so `error` would be emitted on the stream again), I haven't actually run the code but I think the async iterator would (incorrectly) exit in an incorrect state ({finished: true} without `.throw` being called since an error was not thrown from the generator.

Reply:
> so error would be emitted on the stream again

Streams cannot emit error multiple times.

> but I think the async iterator would

You might be right. I will take a look.

Reply:
fixed, PTAL